packages/flutter/lib/src/widgets/form.dart

```diff
@@ -618,13 +640,37 @@ class FormFieldState<T> extends State<FormField<T>> with RestorationMixin {
           if (_hasInteractedByUser.value) {
             _validate();
           }
+        case AutovalidateMode.onUnfocus:
         case AutovalidateMode.disabled:
           break;
       }
     }
+
     Form.maybeOf(context)?._register(this);
+
+    if (Form.maybeOf(context)?.widget.autovalidateMode == AutovalidateMode.onUnfocus && !_isAlwaysAutoValidateMode || widget.autovalidateMode == AutovalidateMode.onUnfocus) {
+      return Focus(
+        canRequestFocus: false,
+        skipTraversal: true,
+        onFocusChange: (bool value) {
+          if (!value) {
+            setState(() {
+              _validate();
+            });
+          }
+        },
+        focusNode: _focusNode,
+        child: widget.builder(this),
+      );
+    }
+
     return widget.builder(this);
   }
+
+  // Returns true if the form field is in a mode where it should always auto validate.
+  // This is used to determine if the form field should listen to focus changes to
+  // trigger validation.
+  bool get _isAlwaysAutoValidateMode => widget.autovalidateMode == AutovalidateMode.always;
```

Comment:
Nit: I think it's weird that this is used only in one place, and you also do `widget.autovalidateMode == AutovalidateMode.onUnfocus` on that line (why is this a getter but that one isn't?). Maybe declare both of them as variables just above line 651? Or inline this one too.

Reply:
I will add inline.